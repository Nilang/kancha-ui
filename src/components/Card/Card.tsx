import * as React from 'react'
import { TouchableHighlight, ViewStyle } from 'react-native'
import Container from '../Container/Container'
import * as Kancha from '../../types'
import { withTheme } from '../../theming'

interface CardProps {
  onPress?: () => void
  shadow?: number
  br?: number
  testID?: string
  background?: Kancha.BrandPropOptions
  theme: any
}

const Card: React.FC<CardProps> = ({ children, onPress, shadow, br, background, testID }) => {
  const style: ViewStyle = {
    ...(shadow
      ? {
          elevation: shadow,
          shadowColor: '#000000',
          shadowOpacity: shadow / 15,
          shadowRadius: shadow * 4,
        }
      : {}),
  }
  return (
    <TouchableHighlight
      onPress={onPress}
      disabled={!onPress}
      testID={testID}
      underlayColor={'transparent'}
      style={style}
    >
      <Container padding={true} br={br ? br : 5} marginBottom={true} background={background}>
        {children}
      </Container>
    </TouchableHighlight>
  )
}

export default withTheme(Card)
