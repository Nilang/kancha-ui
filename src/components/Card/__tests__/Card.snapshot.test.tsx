import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import Card from '../Card'

describe('Component(snapshot): Card', () => {
  it('should render with default props', () => {
    const tree = render(<Card />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const onPress = jest.fn()
    const tree = render(<Card onPress={onPress} shadow={0.3} br={10} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should fire tap event on card when onPress is passed', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(<Card onPress={onPress} />)

    fireEvent.press(getByTestId('CARD_BTN'))
    expect(onPress).toHaveBeenCalled()
  })
})
