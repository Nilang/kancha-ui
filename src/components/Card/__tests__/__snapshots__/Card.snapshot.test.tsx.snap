// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Card should render with default props 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": 5,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Component(snapshot): Card should render with props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "elevation": 0.3,
      "shadowColor": "#000000",
      "shadowOpacity": 0.015,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": 10,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;
